app/main.py

```diff
@@ -1,2 +1,76 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self,
+                 initial_capacity: int = 8,
+                 load_factor: float = 0.75
+                 ) -> None:
+        self.initial_capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.capacity = initial_capacity
+        self.table = [None] * self.capacity
+
+    def __setitem__(self,
+                    key: int | float | str | bool | tuple,
+                    value: Any) -> None:
+        index = self._get_index(key)
+        if self.table[index] is None:
+            self.table[index] = []
+        nodes = self.table[index]
+        for node in nodes:
+            if node[0] == key:
+                node[2] = value
+                return
+        nodes.append([key, self._hash(key), value])
+        self.size += 1
+        self._check_resize()
+
+    def __getitem__(self,
+                    key: int | float | str | bool | tuple
+                    ) -> Any:
+        index = self._get_index(key)
+        nodes = self.table[index]
+        if nodes is not None:
+            for node in nodes:
+                if node[0] == key:
+                    return node[2]
+        raise KeyError(key)
+
+    def __len__(self) -> int:
+        return self.size
+
+    def _get_index(self,
+                   key: int | float | str | bool | tuple
+                   ) -> int:
+        hash_value = hash(key)
+        return hash_value % self.capacity
+
+    @staticmethod
+    def _hash(key: Any) -> int:
```

Comment:
If you'll add Hashable in type annotations, you could remove this function. It will raise TypeError with message "unhashable type: 'list'", for example for list
Which is the same as in the dict